substrate/client/network/sync/src/strategy/state.rs

```diff
@@ -173,7 +173,7 @@ impl<B: BlockT> StateStrategy<B> {
 		peer_id: PeerId,
 		announce: &BlockAnnounce<B::Header>,
 	) -> Option<(B::Hash, NumberFor<B>)> {
-		is_best.then_some({
+		is_best.then(|| {
```

Comment:
This one is pretty bad, it makes `on_validated_block_announce` behave as if `is_best` was always `true`